src/NuGet.Services.Metadata.Catalog.Monitoring/Status/PackageMonitoringStatusService.cs

```diff
@@ -28,40 +26,74 @@ public class PackageMonitoringStatusService : IPackageMonitoringStatusService
         /// </summary>
         private IStorageFactory _storageFactory;
 
-        /// <summary>
-        /// The <see cref="JsonSerializerSettings"/> to use to save and load statuses of packages.
-        /// </summary>
-        private static JsonSerializerSettings SerializerSettings => _serializerSettings.Value;
-        private static Lazy<JsonSerializerSettings> _serializerSettings = new Lazy<JsonSerializerSettings>(() =>
+        public PackageMonitoringStatusService(IStorageFactory storageFactory, ILogger<PackageMonitoringStatusService> logger)
         {
-            var settings = new JsonSerializerSettings();
+            _logger = logger;
+            _storageFactory = storageFactory;
+        }
 
-            settings.Converters.Add(new NuGetVersionConverter());
-            settings.Converters.Add(new StringEnumConverter());
+        public async Task<IEnumerable<PackageMonitoringStatusListItem>> ListAsync(CancellationToken token)
+        {
+            var listTasks =
+                Enum.GetNames(typeof(PackageState))
```

Comment:
`GetNames` uses reflection and is an expensive call. Cache this in the constructor.

Reply:
done, and also cached `Enum.GetValues`